src/server/raft_client.rs

```diff
@@ -16,26 +16,29 @@ use std::net::SocketAddr;
 
 use futures::sync::mpsc::{self, UnboundedSender};
 use futures::sync::oneshot::{self, Sender};
-use futures::{Future, Sink, Stream};
+use futures::{Future, Sink, Stream, stream};
 use grpc::{Environment, ChannelBuilder, WriteFlags};
 use kvproto::raft_serverpb::RaftMessage;
 use kvproto::tikvpb_grpc::TikvClient;
 
 const MAX_GRPC_RECV_MSG_LEN: usize = 10 * 1024 * 1024;
 const MAX_GRPC_SEND_MSG_LEN: usize = 10 * 1024 * 1024;
+const INITIAL_BUFFER_CAP: usize = 1024;
 
 use util::collections::HashMap;
 use super::{Error, Result, Config};
 
 struct Conn {
+    stream: UnboundedSender<Vec<(RaftMessage, WriteFlags)>>,
+    buffer: Vec<(RaftMessage, WriteFlags)>,
+    store_id: u64,
```

Comment:
For removing addrs. https://github.com/pingcap/tikv/pull/2022/files#diff-e60323e8b85aa18ccb78ac822ebb41deR127